features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt

```diff
@@ -500,7 +517,22 @@ class SearchFragment : Fragment(R.layout.fragment_search) {
                 viewModel.logEvent(AnalyticsConstants.Explore.MERCHANT_DETAILS_OPEN_WEBSITE)
             }
         }
-        binding.itemDetails.setOnBuyGiftCardButtonClicked { viewModel.logEvent(AnalyticsConstants.Explore.MERCHANT_DETAILS_BUY_GIFT_CARD) }
+
+        binding.itemDetails.setOnBuyGiftCardButtonClicked {
+            if (!viewModel.isUserSignInDashDirect()) {
+                showLoginDialog()
+            } else {
+                Toast.makeText(requireContext(), "Open Buy Card", Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        viewModel.dashDirectSignIn.observe(viewLifecycleOwner) {
+            if (it) {
+                // TODO open Buy card UI
+                Toast.makeText(requireContext(), "Open Buy Card", Toast.LENGTH_SHORT).show()
+                viewModel.logEvent(AnalyticsConstants.Explore.MERCHANT_DETAILS_BUY_GIFT_CARD)
+            }
+        }
```

Comment:
The pattern of calling a function and observing result in `LiveData` is not great. It makes the execution flow more complicated and hard to track.

Instead, simply call a suspend function in a `lifecycleScope` and process the result.